### PR TITLE
Feat/RTENU-170 favorite category api

### PR DESCRIPTION
**Pull request by NinaDang97**
1. List user's favorite pages
- GET /api/database/favorites
2. Add favorite page to user's list
- POST /api/database/favorites?category=linjakaaviot
3. Remove favorite page from user's list
- DELETE /api/database/favorites?category=linjakaaviot

### CODE REVIEW

**Review by saneDG — APPROVED** [review]

File: packages/server/lambdas/database/delete-favorite-page.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @param {ALBEvent} event
+ * @param {{string}} event.path Path should end with the page to get the custom content for
+ * @returns  {Promise<ALBResult>} JSON stringified object of contents inside body
+ */
+export async function handleRequest(event: ALBEvent): Promise<ALBResult> {
+  try {
+    const user = await getUser(event);
+    const params = event.queryStringParameters;
+    const category = params?.category;
+    log.info(user, 'Delete favorite category page');
```
Comment (mehtis) [inline packages/server/lambdas/database/delete-favorite-page.ts:25]: I'dd log the category here as well.

File: packages/server/lambdas/database/get-favorite-pages.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    log.info(user, 'List favorite category pages');
+    validateReadUser(user);
+
+    const data = await database.favoriteCategory.findMany({
+      where: {
+        userId: user.uid,
+      },
+      select: {
+        id: true,
+        userId: true, // TODO: Remove after testing
```
Comment (mehtis) [inline packages/server/lambdas/database/get-favorite-pages.ts:28]: TODO?

**Review by mehtis — APPROVED** [review]
Only small stuff, so pre-emptive approval already.

File: packages/server/lambdas/database/get-favorite-pages.ts (hunk 2 above)
Comment (NinaDang97) [reply packages/server/lambdas/database/get-favorite-pages.ts:28]: Is it bad if api returns user's id? If not, I could remove comment

Comment (mehtis) [reply packages/server/lambdas/database/get-favorite-pages.ts:28]: The user should know their id, so it's not a big issue. But if it's not needed, better to leave it out.

**Review by noravux — APPROVED** [review]

**Pull request merged**